xmake/rules/qt/xmake.lua

```diff
@@ -121,50 +121,11 @@ rule("qt.widgetapp_static")
     end)
 
     on_config(function (target)
-
-        -- get qt sdk version
-        local qt = target:data("qt")
-        local qt_sdkver = nil
-        if qt.sdkver then
-            import("core.base.semver")
-            qt_sdkver = semver.new(qt.sdkver)
-        end
-
-        -- @see
-        -- https://github.com/xmake-io/xmake/issues/1047
-        -- https://github.com/xmake-io/xmake/issues/2791
-        local QtPlatformSupport
-        if qt_sdkver then
-            if qt_sdkver:ge("6.0") then
-                QtPlatformSupport = nil
-            elseif qt_sdkver:ge("5.9") then
-                QtPlatformSupport = "QtPlatformCompositorSupport"
-            else
-                QtPlatformSupport = "QtPlatformSupport"
-            end
-        end
-
-        -- load some basic plugins and frameworks
-        local plugins = {}
-        local frameworks = {"QtGui", "QtWidgets", "QtCore"}
-        if qt_sdkver and qt_sdkver:lt("5.0") then
-            frameworks = {"QtGui", "QtCore"} -- qt4.x has not QtWidgets, it is in QtGui
-        end
-        if target:is_plat("macosx") then
-            plugins.QCocoaIntegrationPlugin = {linkdirs = "plugins/platforms", links = {"qcocoa", "cups"}}
-            table.insert(frameworks, "QtWidgets")
-            if QtPlatformSupport then
-                table.insert(frameworks, QtPlatformSupport)
-            end
-        elseif target:is_plat("windows") then
-            plugins.QWindowsIntegrationPlugin = {linkdirs = "plugins/platforms", links = {is_mode("debug") and "qwindowsd" or "qwindows"}}
-            table.join2(frameworks, "QtPrintSupport", "QtWidgets")
-            if QtPlatformSupport then
-                table.insert(frameworks, QtPlatformSupport)
-            end
-        elseif target:is_plat("wasm") then
-            plugins.QWasmIntegrationPlugin = {linkdirs = "plugins/platforms", links = {"qwasm"}}
-            table.join2(frameworks, "QtEventDispatcherSupport", "QtFontDatabaseSupport", "QtEglSupport")
```

Comment:
? I don't see where config_static is used. Please revert the patch this time to keep changes as minimal as possible.